### PR TITLE
Reimplement sugar.DSN with net/url

### PR DESCRIPTION
**Pull request by positiveviking**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Reimplement sugar.DSN with net/url
Accidentally trim trailing spaces in CHANGELOG.md

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

File: sugar/dsn.go [hunk 1]
```diff
@@ -1,32 +1,28 @@
...
 
-// DSN makes connection string (data source name) by endpoint, database and secure
-func DSN(endpoint, database string, secure bool) (s string) {
-	cs := connectionString{
-		endpoint: endpoint,
-		database: database,
-		secure:   secure,
+	if secure {
+		dsn.Scheme = "grpcs"
 	}
```
Comment (polRk) [inline sugar/dsn.go:25]: It makes sense to make this default value true.
```suggestion
	if !secure {
	    dsn.Scheme = "grpc"
	}
```

File: sugar/dsn.go [hunk 2]
```diff
@@ -1,32 +1,28 @@
...
 //   sugar.DSN("endpoint", "database", false),
 // )
 
-type connectionString struct {
-	endpoint string
-	database string
-	secure   bool
-}
+// DSN makes connection string (data source name) by endpoint, database and secure
+func DSN(endpoint, database string, secure bool) (s string) {
```
[the remark this answers is not in the document]
Comment (positiveviking) [reply sugar/dsn.go:13]: not in this function.
and for now database is the only parameter

File: sugar/dsn.go (hunk 1 above)
Comment (positiveviking) [reply sugar/dsn.go:25]: here there are no default value this is required parameter

File: options.go [hunk 3]
```diff
@@ -61,28 +62,17 @@ func WithRequestsType(requestsType string) Option {
 // Warning: WithConnectionString will be removed at next major release
 // (connection string will be required string param of ydb.Open)
 func WithConnectionString(connectionString string) Option {
+	if connectionString == "" {
+		return nil
+	}
 	return func(ctx context.Context, c *connection) error {
```
Comment (positiveviking) [inline options.go:68]:
```suggestion
	return func(ctx context.Context, c *connection) error {
		if connectionString == "" {
			return nil
		}
```

И не надо пропуски nil делать в опциях

**Pull request merged**